components/apimgt/org.wso2.carbon.apimgt.rest.api.service.catalog/src/main/java/org/wso2/carbon/apimgt/rest/api/service/catalog/impl/ServicesApiServiceImpl.java

```diff
@@ -115,16 +115,27 @@ public Response deleteService(String serviceId, MessageContext messageContext) {
         String userName = RestApiCommonUtil.getLoggedInUsername();
         int tenantId = APIUtil.getTenantId(userName);
         try {
+            ServiceEntry existingService = serviceCatalog.getServiceByUUID(serviceId, tenantId);
             List<API> usedAPIs = serviceCatalog.getServiceUsage(serviceId, tenantId);
             if (usedAPIs != null && usedAPIs.size() > 0 ) {
                 String message = "Cannot remove the Service as it is used by one or more APIs";
                 RestApiUtil.handleConflict(message, log);
             }
+
+            String uuid = existingService.getUuid();
```

Comment:
existingService can be null from line 118. that will throw a NPE here. handle NPE